pkg/scheduler/plugins/reservation/hook.go

```diff
@@ -108,20 +109,27 @@ func (h *Hook) FilterHook(handle frameworkext.ExtendedHandle, cycleState *framew
 
 	// only continue if matchedCache is prepared in PreFilterHook (i.e. find any reservation matched)
 	state := getPreFilterState(cycleState)
-	if state == nil || state.skip {
+	if state == nil {
 		return nil, nil, false
 	}
+	newNodeInfo := nodeInfo.Clone()
+	if allocatedResource, ok := state.allocatedResources[node.Name]; ok {
+		newNodeInfo.Requested.Add(quotav1.Subtract(util.NewZeroResourceList(), allocatedResource))
```

Comment:
Why is it necessary to add the allocated reserved resources? The former implementation already subtracts the node request with the pod request in `prepareFilterNodeInfo()`.

Reply:
We only subtracts the resource when there are some reservations matched on node.
When the pod matches no reservation on node, and there are reservations which have allocated resources for pods, available resources will be wrong.

Reply:
OK, got it. In summary, we should subtract the allocated reserved resources (or the requests of scheduled pods allocating reserved resources) when there is no reservation matching the pod, and subtract the total reserved when there is any reservation matching the pod. How about doing the subtraction in `prepareFilterNodeInfo()` if we want to change the NodeInfo?

Reply:
Unfortunately, if there is no reservation matching the pod, Skip attribute in prefilterState will be true, and then FilterHook will skip the prepareFilterNodeInfo().
That''s why I put this code here.

Reply:
We can simply change the order between `prepareFilterNodeInfo` and the check of `preFilterState.skip`.